src/sonic-config-engine/sonic_device_util.py

```diff
@@ -64,6 +70,55 @@ def get_namespaces():
         ns_list.append(ns)
     return natsorted(ns_list)
 
+def get_hwsku():
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    metadata = config_db.get_table('DEVICE_METADATA')
+    return metadata['localhost']['hwsku']
+
+def get_platform():
+    if not os.path.isfile('/host/machine.conf'):
+        return ''
+ 
+    with open('/host/machine.conf') as machine_conf:
+        for line in machine_conf:
+            tokens = line.split('=')
+            if tokens[0].strip() == 'onie_platform' or tokens[0].strip() == 'aboot_platform':
+                return tokens[1].strip()
+    return ''
+
+def is_multi_npu():
+    num_npus = get_num_npus()
+    if num_npus > 1:
+        return True
+    else:
+        return False
+
+def get_all_namespaces():
+    """
+    In case of Multi-Asic platform, Each ASIC will have a linux network namespace created.
+    So we loop through the databases in different namespaces and depending on the sub_role
+    decide whether this is a front end ASIC/namespace or a back end one.
+    """
+    front_ns = []
+    back_ns = []
+    num_npus = get_num_npus()
+    SonicDBConfig.load_sonic_global_db_config()
+    
+    if is_multi_npu():
+        for npu in range(num_npus):
+            namespace = "{}{}".format(NPU_NAME_PREFIX, npu)
+            config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+            config_db.connect()
+
+            metadata = config_db.get_table('DEVICE_METADATA')
+            if metadata['localhost']['sub_role'] == 'FrontEnd':
```

Comment:
We should probably make `FrontEnd` and `BackEnd` macros like we did for minigraph.py.

Reply:
@daall Done